Ender Fluid Link Cover GUI won't open in multiplayer

### GregTech CEu Version

v2.7.2

### Modpack Used

_No response_

### Addons Installed

_No response_

### Environment

Forge Server

### New Worlds

Yes

### Cross-Mod Interaction

No

### Expected Behavior

Right click on Ender Fluid Link Cover with screwdriver to open GUI

### Actual Behavior

- Nothing happens on the client side and the GUI doesn't open
- Errors are thrown on the server side

### Steps to Reproduce

1. Put Ender Fluid Link Cover on any fluid handling machine
2. Right click on EnderFluidLinkCover with screwdriver

### Additional Information

```
[08:49:16] [Server thread/FATAL] [net.minecraft.server.MinecraftServer]: Error executing task
java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: net/minecraft/client/gui/GuiPageButtonList$GuiResponder
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_372]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_372]
	at net.minecraft.util.Util.runTask(Util.java:51) [Util.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:797) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:415) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:742) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:591) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_372]
Caused by: java.lang.NoClassDefFoundError: net/minecraft/client/gui/GuiPageButtonList$GuiResponder
	at gregtech.common.covers.CoverEnderFluidLink.createUI(CoverEnderFluidLink.java:157) ~[CoverEnderFluidLink.class:?]
	at gregtech.api.cover.CoverBehaviorUIFactory.createUITemplate(CoverBehaviorUIFactory.java:28) ~[CoverBehaviorUIFactory.class:?]
	at gregtech.api.cover.CoverBehaviorUIFactory.createUITemplate(CoverBehaviorUIFactory.java:15) ~[CoverBehaviorUIFactory.class:?]
	at gregtech.api.gui.UIFactory.openUI(UIFactory.java:35) ~[UIFactory.class:?]
	at gregtech.common.covers.CoverEnderFluidLink.openUI(CoverEnderFluidLink.java:143) ~[CoverEnderFluidLink.class:?]
	at gregtech.common.covers.CoverEnderFluidLink.onScrewdriverClick(CoverEnderFluidLink.java:93) ~[CoverEnderFluidLink.class:?]
	at gregtech.api.metatileentity.MetaTileEntity.onToolClick(MetaTileEntity.java:481) ~[MetaTileEntity.class:?]
	at gregtech.api.block.machines.BlockMachine.onBlockActivated(BlockMachine.java:322) ~[BlockMachine.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.processRightClickBlock(PlayerInteractionManager.java:475) ~[PlayerInteractionManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processTryUseItemOnBlock(NetHandlerPlayServer.java:769) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:68) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:13) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:15) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_372]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_372]
	at net.minecraft.util.Util.runTask(Util.java:50) ~[Util.class:?]
	... 5 more
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.gui.GuiPageButtonList$GuiResponder
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:101) ~[launchwrapper-1.12.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:419) ~[?:1.8.0_372]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352) ~[?:1.8.0_372]
	at gregtech.common.covers.CoverEnderFluidLink.createUI(CoverEnderFluidLink.java:157) ~[CoverEnderFluidLink.class:?]
	at gregtech.api.cover.CoverBehaviorUIFactory.createUITemplate(CoverBehaviorUIFactory.java:28) ~[CoverBehaviorUIFactory.class:?]
	at gregtech.api.cover.CoverBehaviorUIFactory.createUITemplate(CoverBehaviorUIFactory.java:15) ~[CoverBehaviorUIFactory.class:?]
	at gregtech.api.gui.UIFactory.openUI(UIFactory.java:35) ~[UIFactory.class:?]
	at gregtech.common.covers.CoverEnderFluidLink.openUI(CoverEnderFluidLink.java:143) ~[CoverEnderFluidLink.class:?]
	at gregtech.common.covers.CoverEnderFluidLink.onScrewdriverClick(CoverEnderFluidLink.java:93) ~[CoverEnderFluidLink.class:?]
	at gregtech.api.metatileentity.MetaTileEntity.onToolClick(MetaTileEntity.java:481) ~[MetaTileEntity.class:?]
	at gregtech.api.block.machines.BlockMachine.onBlockActivated(BlockMachine.java:322) ~[BlockMachine.class:?]
	at net.minecraft.server.management.PlayerInteractionManager.processRightClickBlock(PlayerInteractionManager.java:475) ~[PlayerInteractionManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processTryUseItemOnBlock(NetHandlerPlayServer.java:769) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:68) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.processPacket(CPacketPlayerTryUseItemOnBlock.java:13) ~[CPacketPlayerTryUseItemOnBlock.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:15) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_372]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_372]
	at net.minecraft.util.Util.runTask(Util.java:50) ~[Util.class:?]
	... 5 more
```

--- a/src/main/java/gregtech/api/gui/widgets/TextFieldWidget.java
+++ b/src/main/java/gregtech/api/gui/widgets/TextFieldWidget.java
@@ -3,11 +3,11 @@ package gregtech.api.gui.widgets;
 import gregtech.api.gui.IRenderContext;
 import gregtech.api.gui.Widget;
 import gregtech.api.gui.resources.IGuiTexture;
+import gregtech.api.util.MCGuiUtil;
 import gregtech.api.util.Position;
 import gregtech.api.util.Size;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.FontRenderer;
-import net.minecraft.client.gui.GuiPageButtonList;
 import net.minecraft.client.gui.GuiTextField;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.network.PacketBuffer;
@@ -45,7 +45,7 @@ public class TextFieldWidget extends Widget {
             this.textField.setCanLoseFocus(true);
             this.textField.setEnableBackgroundDrawing(enableBackground);
             this.textField.setMaxStringLength(this.maxStringLength);
-            this.textField.setGuiResponder(new Responder());
+            this.textField.setGuiResponder(MCGuiUtil.createTextFieldResponder(this::onTextChanged));
         }
         this.textSupplier = textSupplier;
         this.textResponder = textResponder;
@@ -64,7 +64,7 @@ public class TextFieldWidget extends Widget {
             this.textField.setEnableBackgroundDrawing(enableBackground);
             this.textField.setMaxStringLength(maxStringLength);
             this.maxStringLength = maxStringLength;
-            this.textField.setGuiResponder(new Responder());
+            this.textField.setGuiResponder(MCGuiUtil.createTextFieldResponder(this::onTextChanged));
         }
         this.textSupplier = textSupplier;
         this.textResponder = textResponder;
@@ -79,7 +79,7 @@ public class TextFieldWidget extends Widget {
             this.textField.setCanLoseFocus(true);
             this.textField.setEnableBackgroundDrawing(false);
             this.textField.setMaxStringLength(maxStringLength);
-            this.textField.setGuiResponder(new Responder());
+            this.textField.setGuiResponder(MCGuiUtil.createTextFieldResponder(this::onTextChanged));
         }
         this.background = background;
         this.textSupplier = textSupplier;
@@ -236,19 +236,5 @@ public class TextFieldWidget extends Widget {
             this.textField.setValidator(validator::test);
         }
         return this;
-    }
-
-    private class Responder implements GuiPageButtonList.GuiResponder {
-
-        @Override
-        public void setEntryValue(int id, boolean value) {}
-
-        @Override
-        public void setEntryValue(int id, float value) {}
-
-        @Override
-        public void setEntryValue(int id, String value) {
-            onTextChanged(value);
-        }
     }
 }

--- a/src/main/java/gregtech/api/util/MCGuiUtil.java
+++ b/src/main/java/gregtech/api/util/MCGuiUtil.java
@@ -1,0 +1,28 @@
+package gregtech.api.util;
+
+import net.minecraft.client.gui.GuiPageButtonList.GuiResponder;
+
+import javax.annotation.Nonnull;
+import java.util.function.Consumer;
+
+public class MCGuiUtil {
+
+    public static GuiResponder createTextFieldResponder(Consumer<String> onChanged) {
+        return new GuiResponder() {
+            @Override
+            public void setEntryValue(int id, boolean value) {
+            }
+
+            @Override
+            public void setEntryValue(int id, float value) {
+            }
+
+            @Override
+            public void setEntryValue(int id, @Nonnull String value) {
+                onChanged.accept(value);
+            }
+        };
+    }
+
+
+}

--- a/src/main/java/gregtech/api/util/MCGuiUtil.java
+++ b/src/main/java/gregtech/api/util/MCGuiUtil.java
@@ -5,6 +5,10 @@ import net.minecraft.client.gui.GuiPageButtonList.GuiResponder;
 import javax.annotation.Nonnull;
 import java.util.function.Consumer;
 
+/**
+ * This class exists to avoid java always trying to load client classes when loading {@link gregtech.api.gui.widgets.TextFieldWidget}.
+ * Do not remove
+ */
 public class MCGuiUtil {
 
     public static GuiResponder createTextFieldResponder(Consumer<String> onChanged) {

